sql/core/src/test/scala/org/apache/spark/sql/streaming/resources/testCommitLogV1/testCommitLog

```diff
@@ -0,0 +1,2 @@
+v1
+{"nextBatchWatermarkMs":1,"stateUniqueIds":{}}
```

Comment:
It's better to put it as a constant string in the source code and write it in a file when the tests start. I am not sure we can always assume that when the test runs, we can access to the source code files. It is less flexible.

Reply:
This is the golden file generation mechanism, a pretty common pattern used in Spark: 
https://github.com/apache/spark/blob/master/core/src/test/scala/org/apache/spark/SparkFunSuite.scala#L91

See example usage here: https://github.com/apache/spark/blob/master/core/src/test/scala/org/apache/spark/SparkThrowableSuite.scala#L95

Reply:
I assume this shouldn't have the `"stateUniqueIds":{}` field and should continue work without it

Reply:
This is for same-version verification, e.g. a spark 4.0 query (with this change) but have checkpoint format v2 turned off.

For cross version validation (a spark 4.0 query run against commit log created in spark 3.5, see the test here)
```
// Old metadata structure with no state unique ids should not affect the deserialization
  test("Cross-version V1 SerDe") {
    val commitlogV1 = """v1
                        |{"nextBatchWatermarkMs":233}""".stripMargin
    val inputStream: ByteArrayInputStream =
      new ByteArrayInputStream(commitlogV1.getBytes("UTF-8"))
    val commitMetadata: CommitMetadata = new CommitLog(
      spark, testCommitLogV1FilePath.toString).deserialize(inputStream)
    assert(commitMetadata.nextBatchWatermarkMs === 233)
    assert(commitMetadata.stateUniqueIds === Map.empty)
  }
```

Reply:
ah my bad - was worried when I didn't see a golden file for it. Any reason you haven't created a golden file? Then you may not need to expose the methods inside CommitLog?

Reply:
nah just this way is easier...